criu/fsnotify.c

```diff
@@ -602,6 +606,31 @@ static int restore_one_inotify(int inotify_fd, struct fsnotify_mark_info *info)
 
 err:
 	close_safe(&target);
+
+	/*
+	 * As we close target after setting up an inotify on it, we create
+	 * an unwanted inotify event, we need to remove these event.
+	 */
+	if (!ret && (mask & IN_CLOSE_NOWRITE)) {
+		char buff[EVENT_BUF_SIZE];
+		struct inotify_event *ie = (struct inotify_event *)buff;
+		int ielen;
+		int len;
+
+		len = read(inotify_fd, buff, EVENT_BUF_SIZE);
+		if (len < 0) {
+			pr_perror("Failed to read event from inotify %#x", inotify_fd);
+			return -1;
+		}
+
+		ielen = EVENT_SIZE + ie->len;
+		if ((len != ielen) || (ie->mask != IN_CLOSE_NOWRITE)) {
+			pr_perror("Only one close ievent expected, len %d != %d on %#x",
```

Comment:
we need to have two separate errors:
- received more than one event
- the event has a wrong type

Reply:
done.